app/src/main/java/edu/wisc/ece/pinpoint/pages/newpin/NewPinFragment.java

```diff
@@ -0,0 +1,137 @@
+
+import edu.wisc.ece.pinpoint.R;
+import edu.wisc.ece.pinpoint.data.Pin;
+import edu.wisc.ece.pinpoint.data.Pin.PinType;
+import edu.wisc.ece.pinpoint.pages.profile.ProfileFragmentAdapter;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Use the {@link NewPinFragment#newInstance} factory method to
+ * create an instance of this fragment.
+ *
+ */
+public class NewPinFragment extends Fragment {
+
+
+    private TabLayout newpin_tabLayout;
+    private ViewPager2 newpin_viewPager;
+
+    private ScrollView newpin_scrollview;
+
+    private EditText newpin_inputeditlayout;
+
+    private Button dropButton;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_new_pin, container, false);
+    }
+
+
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        newpin_tabLayout = requireView().findViewById(R.id.newpin_tab_layout);
+        newpin_viewPager = requireView().findViewById(R.id.newpin_view_pager);
+        newpin_scrollview = requireView().findViewById(R.id.newpin_scrollview);
+        newpin_inputeditlayout = requireView().findViewById(R.id.newpin_inputeditlayout);
+        newpin_tabLayout.addTab(newpin_tabLayout.newTab().setText("Text"));
+        newpin_tabLayout.addTab(newpin_tabLayout.newTab().setText("Image"));
+        NewPinFragmentAdapter fragmentAdapter =
+                new NewPinFragmentAdapter(this.getChildFragmentManager(),
+                        newpin_tabLayout.getTabCount(), getLifecycle());
+        newpin_viewPager.setAdapter(fragmentAdapter);
+
+        dropButton = requireView().findViewById(R.id.drop_pin_button);
+        dropButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createNewPin();
+            }
+        });
+
+        //Force the screen to scroll to the bottom after the caption text input gets focus.
+        //Post delay is required as the scroll does not do anything if the fragment hasn't been pushed
+        //up first, because the base fragment is too small (no scroll needed)
+        newpin_inputeditlayout.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if(hasFocus) {
+                    newpin_scrollview.postDelayed(
+                            new Runnable() {
+                                @Override
+                                public void run() {
+                                    newpin_scrollview.scrollTo(0, newpin_scrollview.getHeight()*2);
+                                }
+                }, 100);
+                }else {
+
+                }
+            }
+        });
```

Comment:
I think if you wrap the inputs with the responsive material input layouts as I mention farther down you might not need this anymore.